Mathlib/Analysis/NormedSpace/FunctionSeries.lean

```diff
@@ -44,6 +44,55 @@ theorem tendstoUniformlyOn_tsum_nat {f : ℕ → β → F} {u : ℕ → ℝ} (hu
       s :=
   fun v hv => tendsto_finset_range.eventually (tendstoUniformlyOn_tsum hu hfu v hv)
 
+/-- An infinite sum of functions with eventually summable sup norm is the uniform limit of its
+partial sums. Version relative to a set, with general index set. -/
+theorem tendstoUniformlyOn_tsum_eventually {ι : Type*} {f : ι → β → F} {u : ι → ℝ}
+    (hu : Summable u) {s : Set β}
+    (hfu : ∃ a : Finset ι, ∀ᵉ (b ⊇ a) (x ∈ s) (n ∉ b), ‖f n x‖ ≤ u n) :
+    TendstoUniformlyOn (fun t x => ∑ n ∈ t, f n x) (fun x => ∑' n, f n x) atTop s := by
+  classical
+  refine tendstoUniformlyOn_iff.2 fun ε εpos => ?_
+  have := (tendsto_order.1 (tendsto_tsum_compl_atTop_zero u)).2 _ εpos
+  simp only [gt_iff_lt, eventually_atTop, ge_iff_le, Finset.le_eq_subset] at *
+  obtain ⟨t, ht⟩ := this
+  obtain ⟨N, hN⟩ := hfu
+  use N ∪ t
+  intro n hn x hx
+  have A : Summable fun n => ‖f n x‖ := by
+    apply Summable.add_compl (s := N) Summable.of_finite
+    apply Summable.of_nonneg_of_le (fun _ ↦ norm_nonneg _) _ (hu.subtype _)
+    simp only [comp_apply, Subtype.forall, Set.mem_compl_iff, Finset.mem_coe]
+    exact hN N subset_rfl x hx
+  rw [dist_eq_norm, ← sum_add_tsum_subtype_compl A.of_norm n, add_sub_cancel_left]
+  have hN2 := hN n (Finset.union_subset_left hn) x hx
+  have ht2 := ht n (Finset.union_subset_right hn)
+  apply lt_of_le_of_lt _ ht2
+  apply (norm_tsum_le_tsum_norm ?_).trans
+  · apply tsum_le_tsum
+    · intro i
+      apply hN2
+      apply i.2
+    · exact (A.subtype _)
+    · exact (hu.subtype _)
+  · exact (A.subtype _)
+
+/-- An infinite sum of functions with eventually summable sup norm is the uniform limit of its
+partial sums. Version relative to a set, with index set `ℕ`. -/
+theorem tendstoUniformlyOn_tsum_nat_eventually {α F : Type*} [NormedAddCommGroup F]
```

Comment:
CAn you multiplicativise?

Reply:
So like `tendstoUniformlyOn_tsum ` it depends on `norm_tsum_le_tsum_norm`, which I also dont think we have a multiplicative version of (I could be wrong). I think adding these versions (where possible) would go outside the scope of this PR.

Reply:
Hmm okay 😦